codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/client/FluentClientGenerator.kt

```diff
@@ -0,0 +1,599 @@
+                newlinePrefix = "//! ",
+            )
+            operations.forEach { operation ->
+                val operationSymbol = symbolProvider.toSymbol(operation)
+                val input = operation.inputShape(model)
+                val baseDerives = symbolProvider.toSymbol(input).expectRustMetadata().derives
+                val derives = baseDerives.derives.intersect(setOf(RuntimeType.Clone)) + RuntimeType.Debug
+                rust(
+                    """
+                    /// Fluent builder constructing a request to `${operationSymbol.name}`.
+                    ///
+                    """,
+                )
+
+                documentShape(operation, model, autoSuppressMissingDocs = false)
+                deprecatedShape(operation)
+                baseDerives.copy(derives = derives).render(this)
+                rustTemplate(
+                    """
+                    pub struct ${operationSymbol.name}#{generics:W} {
+                        handle: std::sync::Arc<super::Handle${generics.inst}>,
+                        inner: #{Inner}
+                    }
+                    """,
+                    "Inner" to input.builderSymbol(symbolProvider),
+                    "client" to clientDep.asType(),
+                    "generics" to generics.decl,
+                    "operation" to operationSymbol,
+                )
+
+                rustBlockTemplate(
+                    "impl${generics.inst} ${operationSymbol.name}${generics.inst} #{bounds:W}",
+                    "client" to clientDep.asType(),
+                    "bounds" to generics.bounds,
+                ) {
+                    val inputType = symbolProvider.toSymbol(operation.inputShape(model))
+                    val outputType = symbolProvider.toSymbol(operation.outputShape(model))
+                    val errorType = operation.errorSymbol(model, symbolProvider, CodegenTarget.CLIENT)
+                    val retryType =
+                        operation.getTrait<RetryPolicyTrait>()?.let { it.getRetryPolicy(runtimeConfig).fullyQualifiedName() } ?: "()"
+
+                    rustTemplate(
+                        """
+                        /// Creates a new `${operationSymbol.name}`.
+                        pub(crate) fn new(handle: std::sync::Arc<super::Handle${generics.inst}>) -> Self {
+                            Self { handle, inner: Default::default() }
+                        }
+
+                        /// Consume this builder, creating a customizable operation that can be modified before being
+                        /// sent. The operation's inner [http::Request] can be modified as well.
+                        pub async fn customize(self) -> Result<
+                            crate::customizable_operation::CustomizableOperation<
+                                #{Operation},
+                                $retryType,
+                                ${generics.inst.trimStart('<').trimEnd('>')}
```

Comment:
Seems like there should be a separate method/property on generics that produces this in the correct format.